mm2src/coins/lp_coins.rs

```diff
@@ -727,15 +745,38 @@ pub struct EthValidateFeeArgs<'a> {
     pub uuid: &'a [u8],
 }
 
+#[derive(Clone, Debug)]
+pub struct WaitForHTLCTxSpendArgs<'a> {
+    pub tx_bytes: &'a [u8],
+    pub secret_hash: &'a [u8],
```

Comment:
I just did this change locally, but I think it's excessive and will add unneeded code so I will not make it part of the patch PR.

This change to a strict type should be done in the negotiation messages where the secret hash gets exchanged instead of here, then the new type should be used everywhere not just in `WaitForHTLCTxSpendArgs`. This will be a big change that should be done in a separate PR where we should consider how deserialization from and to the strict type should be done for backward compatibility.

We already have this check https://github.com/KomodoPlatform/atomicDEX-API/blob/aea18ebc0d250a3f47585a70bb53e2d603c73d89/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L1164
And changing `WaitForHTLCTxSpendArgs::secret_hash` to a strict type will require doing this check again where it's not needed and adding error returns etc...